CONTRIBUTING.md

```diff
@@ -1,12 +1,32 @@
 # Contributing to Pulumi
 
-First, thanks for contributing to Pulumi and helping make it better. We appreciate the help! If you're looking for an issue to start with, we've tagged some issues with the [help-wanted](https://github.com/pulumi/pulumi/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) tag but feel free to pick up any issue that looks interesting to you or fix a bug you stumble across in the course of using Pulumi. No matter the size, we welcome all improvements.
+First, thanks for contributing to Pulumi and helping make it better. We appreciate the help!
+This repository is one of many across the Pulumi ecosystem and we welcome contributions to them all.
 
-For larger features, we'd appreciate it if you open a [new issue](https://github.com/pulumi/pulumi/issues/new) before doing a ton of work to discuss the feature before you start writing a lot of code.
+## Code of Conduct
 
-## Hacking on Pulumi
+Please make sure to read and observe our [Contributor Code of Conduct](./CODE-OF-CONDUCT.md).
 
-To hack on Pulumi, you'll need to get a development environment set up. You'll want to install the following on your machine:
+## Communications
+
+You are welcome to join the [Pulumi Community Slack](https://slack.pulumi.com/) for questions and a community of like-minded folks.
+We discuss features and file bugs on GitHub via [Issues](https://github.com/pulumi/pulumi/issues) as well as [Discussions](https://github.com/pulumi/pulumi/discussions).
+You can read about our [public roadmap](https://github.com/orgs/pulumi/projects/44) on the [Pulumi blog](https://www.pulumi.com/blog/relaunching-pulumis-public-roadmap/).
+
+### Issues
+
+Feel free to pick up any existing issue that looks interesting to you or fix a bug you stumble across while using Pulumi. No matter the size, we welcome all improvements.
+
+### Feature Work
+
+For larger features, we'd appreciate it if you open a [new issue](https://github.com/pulumi/pulumi/issues/new) before doing a ton of work so we can discuss the feature together.
+Please also ensure to browse [current issues](https://github.com/pulumi/pulumi/issues) to make sure your issue is unique, to lighten the triage burden on our maintainers.
+
+## Developing
+
+### Setting up your Pulumi development environment
+
+You'll want to install the following on your machine:
 
 - Go 1.17
 - NodeJS 12.X.X or later
```

Comment:
Is it only Go and NodeJS? I'm noticing Python in the brew call below, so I'm not sure if this is still accurate. For Node, I think we must be on at least 14 by now considering 16 is the current LTS and 12 and 14 are in Maintenance (12's EOL at the end of April), but I could be wrong.